cmd/saml.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2015 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/tsuru/tsuru/auth/saml"
+)
+
+const formPostPage = `<!DOCTYPE html>
+<html>
+<head>
+	<style>
+	body {
+		display: none;
+	}
+	</style>
+</head>
+<body onload="document.frm.submit()">
+	<form method="POST" name="frm" action="{{.url}}">
+		<input type="hidden" name="SAMLRequest" value="{{.saml_request}}" />
+		<input type="submit" value="Go to login" />
+	</form>
+</body>
+</html>
+`
+
+const successSamlCallBackMarkup = `
+	<script>window.close();</script>
+	<h1>Login Successful!</h1>
+	<p>You can close this window now.</p>
+`
+
+const errorSamlCallBackMarkup = `
+	<h1>Login Failed!</h1>
+	<pre>%s</pre>
+`
+
+func SamlCallbackSuccessMessage() string {
+	return successSamlCallBackMarkup
+}
+
+func SamlCallbackFailureMessage() string {
+	return errorSamlCallBackMarkup
+}
+
+func samlRequestId(schemeData map[string]string) string {
+	p := schemeData["request_id"]
+	return fmt.Sprintf("%s", p)
```

Comment:
the `fmt.Sprintf` is unecessary here. You can just return `schemeData["request_id"]`